go/updater/signature.go

```diff
@@ -0,0 +1,104 @@
+// Copyright 2015 Keybase, Inc. All rights reserved. Use of
+// this source code is governed by the included BSD license.
+
+package updater
+
+import (
+	"bytes"
+	"crypto/rand"
+	"encoding/hex"
+	"fmt"
+	"io"
+	"os"
+
+	"github.com/agl/ed25519"
+	"github.com/keybase/client/go/protocol"
+	"github.com/keybase/client/go/saltpack"
+	"github.com/keybase/client/go/saltpack/basic"
+)
+
+func SignArmoredDetachedFile(inFile string, signer saltpack.SigningSecretKey) (string, error) {
+	in, err := os.Open(inFile)
+	if err != nil {
+		return "", err
+	}
+	defer in.Close()
+	return SignArmoredDetached(in, signer)
+}
+
+func SignArmoredDetached(in io.Reader, signer saltpack.SigningSecretKey) (string, error) {
+	var buf bytes.Buffer
+	stream, err := saltpack.NewSignDetachedArmor62Stream(&buf, signer, "KEYBASE")
+	if err != nil {
+		return "", err
+	}
+	if _, err := io.Copy(stream, in); err != nil {
+		return "", err
+	}
+	if err := stream.Close(); err != nil {
+		return "", err
+	}
+	return buf.String(), nil
+}
+
+func VerifyUpdate(update keybase1.Update, publicKeyHex string) error {
+	publicKey, err := decodePublicKeyHex(publicKeyHex)
```

Comment:
Can you use our `KID` import/export functions?  Also if you use the `saltpack_verify` keyring, you'll get this for free.